hazelcast/src/main/java/com/hazelcast/partition/impl/InternalPartitionServiceImpl.java

```diff
@@ -2102,15 +2102,22 @@ public void run() {
         }
 
         private void doRun() throws InterruptedException {
-            for (; ; ) {
+            for (; ;) {
                 if (!isMigrationAllowed()) {
                     break;
                 }
-                Runnable r = migrationQueue.poll(1, TimeUnit.SECONDS);
+
+                Runnable r = migrationQueue.peek();
                 if (r == null) {
+                    Thread.sleep(100);
```

Comment:
Before this change, ``poll()`` is used with 1 second of timeout. So if there is no migration task, it will be checked every 1 second. Once a migration task is put into the queue, it will be polled immediately. I used 100 ms sleep duration to get the migration task without too much delay.